addons/sale_expense/models/account_move_line.py

```diff
@@ -12,7 +12,9 @@ def _sale_can_be_reinvoice(self):
         """
         self.ensure_one()
         if self.expense_id:  # expense flow is different from vendor bill reinvoice flow
-            return self.expense_id.product_id.expense_policy in {'sales_price', 'cost'} and self.expense_id.sale_order_id
+            return self.expense_id.product_id.expense_policy in {'sales_price', 'cost'} \
+                and self.expense_id.sale_order_id \
+                and self.display_type == 'product'
```

Comment:
```suggestion
            return (
                self.expense_id.product_id.expense_policy in {'sales_price', 'cost'}
                and self.expense_id.sale_order_id
                and self.display_type == 'product'
            )
```
https://peps.python.org/pep-0008/#maximum-line-length
> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. ***These should be used in preference to using a backslash for line continuation.***